ansible_collections/arista/avd/plugins/README.md

```diff
@@ -308,6 +323,23 @@ switch:
                         platform_settings | selectattr("platforms", "arista.avd.contains", "default") | first) }}
 ```
 
+### bgp_valid_password test
+
+The `arista.avd.bgp_valid_password` test will test if an encrypted BGP password provided is valid against the Neighbor IP or Peer Group Name against which it is applied.
+It allows to detect early in the configuration generation step if there will be an issue when applying the configuration on the device.
+
+The test accepts either a single value and takes as input the `key` either Neighbor IP or BGP Peer Group Name.
+
+To use this test:
+
+```jinja
+{% if bgp_password is arista.avd.bgp_valid_password(key="My-Awesome-Peer-Group-Name %}
+  neighbor My-Awesome-Peer-Group-Name password bgp_password
+{% endif %}
+```
+
+If the password is not valid, an Exception is raised
```

Comment:
A test should just return false if it is not valid.

Reply:
this is not usable then as you don't get any warning - we are doing it already in our other test btw: https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/plugins/test/defined.py#L84